File: OSESMO MATLAB/OSESMO Option S/Model Code/Import_Load_Profile_Data.m
function [Load_Profile_Data, Load_Profile_Master_Index] = Import_Load_Profile_Data(Input_Output_Data_Directory_Location, OSESMO_Git_Repo_Directory, delta_t, Load_Profile_Name_Input)

% Set Directory to Box Sync Folder
cd(Input_Output_Data_Directory_Location)

% Import Load Profile Data

switch Load_Profile_Name_Input      
        
    case "EnerNOC GreenButton Los Angeles Grocery"
        
        Load_Profile_Master_Index = "C9";
        
        if delta_t == (5/60)
            Load_Profile_Data = csvread(['Load Profile Data/EnerNOC GreenButton/' ...
                'Selected Clean 2017 EnerNOC Load Profiles/5-Minute Data/Los Angeles Grocery/' ...
                'Vector Format/Clean_Vector_2017_Los_Angeles_Grocery.csv']);
        elseif delta_t == (15/60)
            Load_Profile_Data = csvread(['Load Profile Data/EnerNOC GreenButton/' ...
                'Selected Clean 2017 EnerNOC Load Profiles/15-Minute Data/Los Angeles Grocery/' ...
                'Vector Format/Clean_Vector_2017_Los_Angeles_Grocery.csv']);
        end
        
    case "EnerNOC GreenButton Los Angeles Industrial"
        
        Load_Profile_Master_Index = "C10";
        
        if delta_t == (5/60)
            Load_Profile_Data = csvread(['Load Profile Data/EnerNOC GreenButton/' ...
                'Selected Clean 2017 EnerNOC Load Profiles/5-Minute Data/Los Angeles Industrial/' ...
                'Vector Format/Clean_Vector_2017_Los_Angeles_Industrial.csv']);
        elseif delta_t == (15/60)
            Load_Profile_Data = csvread(['Load Profile Data/EnerNOC GreenButton/' ...
                'Selected Clean 2017 EnerNOC Load Profiles/15-Minute Data/Los Angeles Industrial/' ...
                'Vector Format/Clean_Vector_2017_Los_Angeles_Industrial.csv']);
        end
        
    case "EnerNOC GreenButton San Diego Office"
        
        Load_Profile_Master_Index = "C11";
        
        if delta_t == (5/60)
            Load_Profile_Data = csvread(['Load Profile Data/EnerNOC GreenButton/' ...
                'Selected Clean 2017 EnerNOC Load Profiles/5-Minute Data/San Diego Office/' ...
                'Vector Format/Clean_Vector_2017_San_Diego_Office.csv']);
        elseif delta_t == (15/60)
            Load_Profile_Data = csvread(['Load Profile Data/EnerNOC GreenButton/' ...
                'Selected Clean 2017 EnerNOC Load Profiles/15-Minute Data/San Diego Office/' ...
                'Vector Format/Clean_Vector_2017_San_Diego_Office.csv']);
        end
        
        
    case "EnerNOC GreenButton San Francisco Industrial"
        
        Load_Profile_Master_Index = "C12";
        
        if delta_t == (5/60)
            Load_Profile_Data = csvread(['Load Profile Data/EnerNOC GreenButton/' ...
                'Selected Clean 2017 EnerNOC Load Profiles/5-Minute Data/San Francisco Industrial/' ...
                'Vector Format/Clean_Vector_2017_San_Francisco_Industrial.csv']);
        elseif delta_t == (15/60)
            Load_Profile_Data = csvread(['Load Profile Data/EnerNOC GreenButton/' ...
                'Selected Clean 2017 EnerNOC Load Profiles/15-Minute Data/San Francisco Industrial/' ...
                'Vector Format/Clean_Vector_2017_San_Francisco_Industrial.csv']);
        end
        
        
    case "EnerNOC GreenButton San Francisco Office"
        
        Load_Profile_Master_Index = "C13";
        
        if delta_t == (5/60)
            Load_Profile_Data = csvread(['Load Profile Data/EnerNOC GreenButton/' ...
                'Selected Clean 2017 EnerNOC Load Profiles/5-Minute Data/San Francisco Office/' ...
                'Vector Format/Clean_Vector_2017_San_Francisco_Office.csv']);
        elseif delta_t == (15/60)
            Load_Profile_Data = csvread(['Load Profile Data/EnerNOC GreenButton/' ...
                'Selected Clean 2017 EnerNOC Load Profiles/15-Minute Data/San Francisco Office/' ...
                'Vector Format/Clean_Vector_2017_San_Francisco_Office.csv']);
        end
               
        
end

% Return to OSESMO Git Repository Directory
cd(OSESMO_Git_Repo_Directory)

end

File: OSESMO R/Import_Load_Profile_Data.R
Import_Load_Profile_Data <- function(Input_Output_Data_Directory_Location, OSESMO_Git_Repo_Directory, delta_t, Load_Profile_Name_Input){

  # Set Directory to Input/Output Directory
  setwd(Input_Output_Data_Directory_Location)
  
  # Import Load Profile Data
  
  if(Load_Profile_Name_Input == "PG&E GreenButton Central Valley Residential Non-CARE"){
    
    Load_Profile_Master_Index = "R9"
    
    if(delta_t == (60/60)){
      Load_Profile_Data = read.csv(file.path("Load Profile Data", "PG&E Residential Central Valley 2015",
                                             "2017 Remapped", "60-Minute Data",
                                             "Vector Format", "Clean_Vector_2017_PGE_Central_Valley_Residential_Non_CARE.csv"), header = F)[, 1]
    }
    
    
  }
  
  if(Load_Profile_Name_Input == "EnerNOC GreenButton San Francisco Office"){
    
    Load_Profile_Master_Index = "C13"
    
    if(delta_t == (15/60)){
      Load_Profile_Data = read.csv(file.path("Load Profile Data", "EnerNOC GreenButton",
                                             "Selected Clean 2017 EnerNOC Load Profiles", "15-Minute Data/San Francisco Office",
                                             "Vector Format", "Clean_Vector_2017_San_Francisco_Office.csv"), header = F)[, 1]
    }
    
    
  }
  
  # Return to OSESMO Git Repository Directory
  setwd(OSESMO_Git_Repo_Directory)
  
  
  return(list(Load_Profile_Data, Load_Profile_Master_Index))
  
}